Scripts/Level_Components/Level_Elements/LevelControl.gd

```diff
@@ -40,3 +42,22 @@ func _on_DoubleSpeed_toggled(button_pressed):
 		GameStats.run_speed = 0.25
 	else:
 		GameStats.run_speed = 0.50
+
+
+func _on_Help_pressed():
+	var level = get_node("..")
+	var root = get_tree().root
+	
+	if level.has_tutorial:
+		#If there isn't anymore dialogue to display
+		if DialogueManager.dialogue_queue.size() == 0:
+			DialogueManager.restart_dialogue()
+			var levelPath = root.get_child(root.get_child_count() - 1).filename
+			GameStats.playTutorial[levelPath] = true
+			DialogueManager.add_dialogue(level, level.textPath)
```

Comment:
For levels with has_tutorial = true no generic dialogue text will displayed when pressing Help, Help will just replay tutorial dialogue

I could just remove the if dialogue_queue.size == 0 tho so it will restart the tutorial even if it hasn’t been finished entirely yet

Reply:
Highkey idek if restarting the tutorial when it's not done is the best move but idk what else we would do since it just makes it seem like the player presses help and no help is given if they press it when has_tutorial = true